test/lib/RedefineClassHelper.java

```diff
@@ -61,6 +70,28 @@ public static void redefineClass(Class<?> clazz, byte[] bytecode) throws Excepti
         instrumentation.redefineClasses(new ClassDefinition(clazz, bytecode));
     }
 
+    private static byte[] getBytecodes(ClassLoader loader, String name) throws Exception {
+        try(InputStream is = loader.getResourceAsStream(name + ".class")) {
+            byte[] buf = is.readAllBytes();
+            System.out.println("sizeof(" + name + ".class) == " + buf.length);
+            return buf;
+        }
+    }
+
+    /*
+     * Replace class name in bytecodes to the class we're trying to redefine, so that both
+     * old and new classes can be compiled with jtreg for the test.
+     *
+     * @param loader ClassLoader to find the bytes for the class.
+     * @param oldClassName old class name.
+     * @param newClassName new class name to replace with old class name.
+     */
+    public static byte[] replaceAllStrings(ClassLoader loader, String oldClassName, String newClassName) throws Exception {
+        byte[] buf = getBytecodes(loader, oldClassName);
+        ClassModel classModel = ClassFile.of().parse(buf);
+        return ClassFile.of().build(ClassDesc.of(newClassName), classModel::forEach);
```

Comment:
That's cool! I have no idea how to read it to understand what it does, but it is cool it can be done in one line. :)

Seriously though, which part of that is doing the actual replace??

Reply:
Java's gone overly terse, but if you look at the n-1 commit, the classModel::forEach copies everything in the class to the new class.  At least that's what Chen told me and it seems to do.

Reply:
> Seriously though, which part of that is doing the actual replace??

`ClassDesc.of(newClassName)` defines a new class with the new name, and the lambda just copies everything over as-is.

Reply:
Yes, I wanted to use the ClassRemapper but all the tests that used RedefineClassHelper would have had to add @modules.  We can change it later when ClassRemapper becomes available without adding modules.

Reply:
> the lambda just copies everything over as-is.
..
> But be aware this code does not change the class name use anywhere in the code, descriptors, handles, annotation, etc..., just the class name itself.

@coleenp don't we need to class name to be changed everywhere it appears in the classfile? Or do we know these classes are trivial enough that there is only a single occurrence of the name (and everything else is symbolic via the constant pool) ?